kafka-plugins-client/src/main/java/io/cdap/plugin/sink/KafkaOutputFormat.java

```diff
@@ -43,85 +43,84 @@ public class KafkaOutputFormat extends OutputFormat<Text, Text> {
   private KafkaProducer<String, String> producer;
 
   @Override
-  public void checkOutputSpecs(JobContext jobContext) throws IOException, InterruptedException {
+  public void checkOutputSpecs(JobContext jobContext) {
   }
 
   @Override
-  public OutputCommitter getOutputCommitter(TaskAttemptContext taskAttemptContext) throws IOException,
-    InterruptedException {
+  public OutputCommitter getOutputCommitter(TaskAttemptContext taskAttemptContext) {
     return new OutputCommitter() {
       @Override
-      public void setupJob(JobContext jobContext) throws IOException {
+      public void setupJob(JobContext jobContext) {
         // no-op
       }
 
       @Override
-      public void setupTask(TaskAttemptContext taskContext) throws IOException {
+      public void setupTask(TaskAttemptContext taskContext) {
         //no-op
       }
 
       @Override
-      public boolean needsTaskCommit(TaskAttemptContext taskContext) throws IOException {
+      public boolean needsTaskCommit(TaskAttemptContext taskContext) {
         return false;
       }
 
       @Override
-      public void commitTask(TaskAttemptContext taskContext) throws IOException {
+      public void commitTask(TaskAttemptContext taskContext) {
         //no-op
       }
 
       @Override
-      public void abortTask(TaskAttemptContext taskContext) throws IOException {
+      public void abortTask(TaskAttemptContext taskContext) {
         //no-op
       }
     };
   }
 
   @Override
-  public RecordWriter<Text, Text> getRecordWriter(TaskAttemptContext context)
-    throws IOException, InterruptedException {
+  public RecordWriter<Text, Text> getRecordWriter(TaskAttemptContext context) {
     Configuration configuration = context.getConfiguration();
 
     // Extract the topics
     String topic = configuration.get("topic");
 
-    Properties props = new Properties();
-    // Configure the properties for kafka.
-    props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG,
+    // CDAP-9178: cached the producer object to avoid being created on every batch interval
+    if (producer == null) {
```

Comment:
nit: it would be nice if this logic to initialize the producer is moved to a separate function for cleanliness

Reply:
Fixed.